src/request/selectors/RequestDetailDataSelectors.ts

```diff
@@ -1,19 +1,73 @@
 import { IRequestState } from '../stores/IRequestState';
 
 import { createSelector } from 'reselect';
+import * as _ from 'lodash';
 
+export const getFilterState = (state: IRequestState) => state.detail.data.filters;
 export const getOperations = (state: IRequestState) => state.detail.data.operations;
-export const getSelectedIndex = (state: IRequestState) => state.detail.data.selectedIndex;
+export const getSelectedOperationIdState = (state: IRequestState) => state.detail.data.selectedOperationId;
 
 export const getTotalOperationCount = createSelector(
     getOperations,
     operations => {
         return operations.length;
     });
 
+export const getFilteredOperations = createSelector(
+    getFilterState,
+    getOperations,
+    (filterState, operations) => {
+        const hiddenDatabases = _.keys(_.omitBy(filterState, isShown => isShown));
+        
+        let filteredOperations = [];
+        
+        operations.forEach((operation, index) => {
+            if (!_.includes(hiddenDatabases, operation.database)) {
+                filteredOperations.push({
+                    ordinal: index + 1,
+                    operation: operation
+                })
+            }
+        });
+        
+        return filteredOperations;
+    });
```

Comment:
I'm assuming this is deciding what to include and not include based on its index? If so, longer term when we start introducing things like sorting, additional filtering, etc if we want to base this on a fixed id or something.